content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+---
+linktitle: Unbounded Arrays in AZSL
+title: AZSL, Binding Rules For Unbounded Arrays
+description: There are limitations into how many and what kind of Unbounded Arrays can be declared. 
+weight: 100
+---
+AZSL supports declaration of Unbounded Arrays inside `ShaderResourceGroup`s, but there are limitations into how many and what kind of Unbounded Arrays can be declared. The shader compiler, **AZSLc**, makes sure those rules are not violated.  
+  
+Here is a summary of the rules and limitations per platform (DX12, Vulkan and Metal). '--use-spaces' and '--unique-idx' are command line arguments for **AZSLc** that are used for each platform.  
+| --use-spaces   | --unique-idx | platform | Results |
+|----------------|--------------|----------|---------|
+| OFF | OFF |                   | Only the last SRG can contain one last unbounded array for each:<br> t[], u[], b[], s[]|
+| OFF | ON  |                   | Only the last SRG can contain one, and only one, last unbounded array across all resource types:<br>Only one of t[], u[], b[], s[]|
+| ON  | OFF | DX12              | Each SRG can contain one last unbounded array for each: <br> t[], u[], b[], s[]|
+| ON  | ON  | Vulkan<br> Metal  | Each SRG can contain one, and only one, last unbounded array across all resource types: <br> Only one of t[], u[], b[], s[]|
+  
+## Things to know before going over some examples:
+* "Register Space" in HLSL is the same as "Descriptor Set" in Vulkan.
+* When "--use-spaces" is used, **AZSLc** will assign a unique Register Space per SRG.
+* When "--unique-idx" is used, **AZSLc** will emit all resource descriptors in a Register Space to be in numerical sequence, regardless of the descriptor type. Platforms like Vulkan and Metal require this.
+* Unbounded arrays take over the whole range of register indices from the point they are being declared. This means two things:
+    1. It is possible to declare any kind of resources or any amount of bounded arrays of resources, BEFORE any given Unbounded Array of a particular kind.
+    2. Any resource declared AFTER an Unbounded Array will be considered an error.
```

Comment:
Notes for each bullet's edit suggestion:
* Bullet about "register space"
  * Lower case register space and descriptor set. 
  * Italicize "register space" because it is a term that you're defining on this page. 
  * Important: Explain in a sentence or two what register space is? With this, the user can begin to understand from your docs how it is similar to Vulkan. Then, if they wish to learn more, they can conduct further research into Vulkan docs. 
  
    For example, see my suggestion.
* Bullet about `--use-spaces` option    
  * Add context for when we use the `--use-spaces` option. (The context is when we are compiling.)
  * Use in-line code style for command line arguments and options. 
  * Remove bold from AZSLc. Only bold first instance per page. 
  * Lower case register space, as it's written in [Microsoft docs](https://docs.microsoft.com/en-us/windows/win32/direct3d12/resource-binding-in-hlsl)

* Bullet about `--unique-idx` option
  * Same as previous comments.
  * Change "Platforms like Vulkan and Metal..." to "Vulkan and Metal platforms..." to be clear. The phrase, "Platforms like Vulkan and Metal..." indicates that Vulkan and Metal are only some platforms that require this. It makes me  wonder if DirectX requires this too or not? 

* Bullet about unbounded arrays. 
   * Edits to make it read better. 

```suggestion
* In HLSL, a _register space_ is a slot that connects shader data from the CPU to the GPU. For comparison, a register space is similar to a descriptor set in Vulkan. 

* When compiling shader code with the `--use-spaces` option, AZSLc assigns a unique register space per SRG.

* When compiling shader code with the `--unique-idx` option, AZSLc emits all resource descriptors into numerical sequences, regardless of the descriptor type. Vulkan and Metal platforms require this.

* Unbounded arrays take over the whole range of register indices from the point they are declared. This means you must declare any resources including bounded arrays before declaring an unbounded array. Any resource declared after an unbounded array will cause an error.
```

Reply:
Fixed.